common/dynamicconfig/interface.go

```diff
@@ -27,23 +27,31 @@
 package dynamicconfig
 
 import (
+	"os"
 	"time"
 )
 
 // Client allows fetching values from a dynamic configuration system NOTE: This does not have async
 // options right now. In the interest of keeping it minimal, we can add when requirement arises.
-type Client interface {
-	GetValue(name Key, defaultValue interface{}) (interface{}, error)
-	GetValueWithFilters(name Key, filters map[Filter]interface{}, defaultValue interface{}) (interface{}, error)
+type (
+	Client interface {
+		GetValue(name Key, defaultValue interface{}) (interface{}, error)
+		GetValueWithFilters(name Key, filters map[Filter]interface{}, defaultValue interface{}) (interface{}, error)
 
-	GetIntValue(name Key, filters map[Filter]interface{}, defaultValue int) (int, error)
-	GetFloatValue(name Key, filters map[Filter]interface{}, defaultValue float64) (float64, error)
-	GetBoolValue(name Key, filters map[Filter]interface{}, defaultValue bool) (bool, error)
-	GetStringValue(name Key, filters map[Filter]interface{}, defaultValue string) (string, error)
-	GetMapValue(
-		name Key, filters map[Filter]interface{}, defaultValue map[string]interface{},
-	) (map[string]interface{}, error)
-	GetDurationValue(
-		name Key, filters map[Filter]interface{}, defaultValue time.Duration,
-	) (time.Duration, error)
-}
+		GetIntValue(name Key, filters map[Filter]interface{}, defaultValue int) (int, error)
+		GetFloatValue(name Key, filters map[Filter]interface{}, defaultValue float64) (float64, error)
+		GetBoolValue(name Key, filters map[Filter]interface{}, defaultValue bool) (bool, error)
+		GetStringValue(name Key, filters map[Filter]interface{}, defaultValue string) (string, error)
+		GetMapValue(
+			name Key, filters map[Filter]interface{}, defaultValue map[string]interface{},
+		) (map[string]interface{}, error)
+		GetDurationValue(
+			name Key, filters map[Filter]interface{}, defaultValue time.Duration,
+		) (time.Duration, error)
+	}
+
+	FileReader interface {
+		Stat(src string) (os.FileInfo, error)
+		ReadFile(src string) ([]byte, error)
+	}
```

Comment:
This does not need to be part of public interface. This should be internal to file based client.